server.go

```diff
@@ -1316,6 +1320,32 @@ func (s *Server) Authenticate(username, password string) (*User, error) {
 	return u, nil
 }
 
+// Authenticate by secret file
+func (s *Server) AuthenticateBySecret(secret string) (*User, error) {
+	s.mu.RLock()
+	defer s.mu.RUnlock()
+
+	// load content of the file (first line only)
+	file, err := os.Open(s.SecretFile)
```

Comment:
Hmmm I know it looks bad, but the point of having an secret file that you can change it during running time. The idea is that the secret file is used only for local management, so we don't except massive calls via secret file authentication anyway.

Having an changing secret file if needed, gives config management tool an easier way to manipulate with this file without restarting the service. Also if the file has to be defined before the service starts make config management more difficult because you need to make sure that the file is created after the installation of the packages, but before the service get started :)

Don't get me wrong we can get around it, but just wanted to explain my reasoning for that.